### DISCUSSION TITLE
How to develop plugins?

### DISCUSSION THREAD

**Discussion by zgLiuBing** [apache/apisix#5197] [Q&A]
### Issue description

哪里有开发插件的具体步骤文档呀？哪里可以找apisix项目的目录结构详解呀？

### Environment

- apisix version (cmd: `apisix version`):
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- the plugin runner version, if the issue is about a plugin runner (cmd: depended on the kind of runner):
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

**Comment by shuaijinchao** [comment]
You can look at the documents in this directory.

https://github.com/apache/apisix/tree/master/docs/en/latest/architecture-design

**Reply by juzhiyuan** [reply]
see https://apisix.apache.org/docs/apisix/plugin-develop
